xrootd logs stack-trace on malformed request

Part of DESY testing involves running nmap on open ports.  For prometheus, this includes the xrootd door.  The nmap invocation has it attempt to identify the kind of endpoint by sending content and watching the response.

When receiving badly formatted input, xrootd door logs a stack-trace.  The following is an example of this:

    24 Jun 2015 15:12:04 (xrootd) [door:xrootd:AAUZQ0DV9CE] Uncaught exception in thread xrootd-net-0
    io.netty.handler.codec.DecoderException: java.lang.IllegalStateException
            at     io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:358) ~[netty-codec-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:230) ~[netty-codec-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at org.dcache.xrootd.door.NettyXrootdServer$SessionHandler.channelRead(NettyXrootdServer.java:282) [dcache-xrootd-2.14.0-SNAPSHOT.jar:2.14.0-SNAPSHOT]
            at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
            at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:111) [netty-common-4.0.28.Final.jar:4.0.28.Final]
            at org.dcache.util.CDCThreadFactory$1.run(CDCThreadFactory.java:53) [dcache-core-2.14.0-SNAPSHOT.jar:2.14.0-SNAPSHOT]
            at java.lang.Thread.run(Thread.java:745) [na:1.8.0_45]
    Caused by: java.lang.IllegalStateException: null
            at com.google.common.base.Preconditions.checkState(Preconditions.java:158) ~[guava-18.0.jar:na]
            at org.dcache.xrootd.protocol.messages.AbstractXrootdRequest.<init>(AbstractXrootdRequest.java:38) ~[xrootd4j-2.1.0.jar:2.1.0]
            at org.dcache.xrootd.protocol.messages.HandshakeRequest.<init>(HandshakeRequest.java:30) ~[xrootd4j-2.1.0.jar:2.1.0]
            at org.dcache.xrootd.core.XrootdDecoder.decode(XrootdDecoder.java:60) ~[xrootd4j-2.1.0.jar:2.1.0]
            at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:327) ~[netty-codec-4.0.28.Final.jar:4.0.28.Final]
            ... 18 common frames omitted

I actually get a different stack trace:

    01 jul. 2015 08:19:59 (Xrootd-Gerds-MacBook-Pro) [] Failed to set a channel option: [id: 0x08683a50, /0.0.0.224:0 => /0:0:0:0:0:0:7f00:1:1094]
    io.netty.channel.ChannelException: java.net.SocketException: Invalid argument
        at io.netty.channel.socket.DefaultSocketChannelConfig.setTcpNoDelay(DefaultSocketChannelConfig.java:256) ~[netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.socket.DefaultSocketChannelConfig.setOption(DefaultSocketChannelConfig.java:109) ~[netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.bootstrap.ServerBootstrap$ServerBootstrapAcceptor.channelRead(ServerBootstrap.java:238) ~[netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:308) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:294) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:93) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354) [netty-transport-4.0.28.Final.jar:4.0.28.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:111) [netty-common-4.0.28.Final.jar:4.0.28.Final]
        at org.dcache.util.CDCThreadFactory$1.run(CDCThreadFactory.java:53) [dcache-core-2.14.0-SNAPSHOT.jar:2.14.0-SNAPSHOT]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_45]
    Caused by: java.net.SocketException: Invalid argument
        at sun.nio.ch.Net.setIntOption0(Native Method) ~[na:1.8.0_45]
        at sun.nio.ch.Net.setSocketOption(Net.java:356) ~[na:1.8.0_45]
        at sun.nio.ch.SocketChannelImpl.setOption(SocketChannelImpl.java:190) ~[na:1.8.0_45]
        at sun.nio.ch.SocketAdaptor.setBooleanOption(SocketAdaptor.java:271) ~[na:1.8.0_45]
        at sun.nio.ch.SocketAdaptor.setTcpNoDelay(SocketAdaptor.java:306) ~[na:1.8.0_45]
        at io.netty.channel.socket.DefaultSocketChannelConfig.setTcpNoDelay(DefaultSocketChannelConfig.java:254) ~[netty-transport-4.0.28.Final.jar:4.0.28.Final]
        ... 13 common frames omitted

That's a bug too, but can you give me the nmap options to reproduce your stack trace?

--- a/pom.xml
+++ b/pom.xml
@@ -69,7 +69,7 @@
         <version.xerces>2.11.0</version.xerces>
         <version.jetty>9.2.10.v20150310</version.jetty>
         <version.wicket>6.19.0</version.wicket>
-        <version.xrootd4j>2.1.1</version.xrootd4j>
+        <version.xrootd4j>2.1.2</version.xrootd4j>
         <version.jglobus>2.0.6-rc9.d</version.jglobus>
 
         <!-- BouncyCastle seems to change the naming convention of

